kustomize/commands/edit/remove/removeconfigmap_test.go

```diff
@@ -92,16 +96,67 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 `,
 		},
+		"succeeds when one configmap name exists in the specified namespace": {
+			input: `
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+configMapGenerator:
+- name: test-cm
+  namespace: test-ns
+  files:
+  - application.properties
+`,
+			expectedOutput: `
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+`,
+			wantErr: false,
+			args:    []string{"test-cm", "--namespace=test-ns"},
+		},
+		"handles empty namespace as default in the args": {
+			input: `
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+configMapGenerator:
+- name: test-cm
+  namespace: default
+  files:
+  - application.properties
+`,
+			expectedOutput: `
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+`,
+			wantErr: false,
+			args:    []string{"test-cm"},
+		},
+		"handles empty namespace as default in the kustomization file": {
+			input: `
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+configMapGenerator:
+- name: test-cm
+  files:
+  - application.properties
+`,
+			expectedOutput: `
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+`,
+			wantErr: false,
+			args:    []string{"test-cm", "--namespace=default"},
```

Comment:
What do you think about adding a few more tests that cover the following scenarios?

- The `--namespace` option is omitted and a ConfigMap is in a non-default namespace.
- The `--namespace` option is a non-default namespace that is not the same namespace of the ConfigMap in the Kustomization file/
- There are multiple ConfigMaps in the Kustomization file. One ConfigMap matches the `--namespace` option and the other ConfigMap is in a different namespace.

Reply:
Hi, yes! Thanks for flagging this, I'll update the PR soon to include these test cases.

Reply:
Added the extra tests, could you please take another look and check if it makes sense?

Reply:
Thank you for adding more tests 😀 .

/lgtm

Reply:
Thanks for the review, @ncapps! 🙏